js/src/components/DirectedRectEdge.vue

```diff
@@ -0,0 +1,186 @@
+<template>
```

Comment:
This class seems to have a lot in common with the DirectedEdge class. Perhaps it would make sense to have a common base class for DirectedEdge and DirectedRectEdge.